samtranslator/model/api/api_generator.py

```diff
@@ -1326,8 +1326,19 @@ def _set_endpoint_configuration(self, rest_api: ApiGatewayRestApi, value: Union[
         if isinstance(value, dict) and value.get("Type"):
             rest_api.Parameters = {"endpointConfigurationTypes": value.get("Type")}
             rest_api.EndpointConfiguration = {"Types": [value.get("Type")]}
-            if "VPCEndpointIds" in value:
-                rest_api.EndpointConfiguration["VpcEndpointIds"] = value.get("VPCEndpointIds")
+
+            # SAM API `EndpointConfiguration` uses `VPCEndpointIds` but APIGateway RestApi uses `VpcEndpointIds`.
+            # Deny when both properties are defined at the same time.
```

Comment:
nit: a small sentence explaing why we accept both `VPCEndpointIds` and `VpcEndpointIds` would help